Source/ACE/Network/Handlers/CharacterHandler.cs

```diff
@@ -124,33 +124,107 @@ public static void CharacterRestore(ClientMessage message, Session session)
             reader.Skip(4);   /* Unknown constant (1) */
             character.Heritage = reader.ReadUInt32();
             character.Gender = reader.ReadUInt32();
-            Appearance appearance = Appearance.FromNetowrk(reader);
+            Appearance appearance = Appearance.FromNetwork(reader);
 
             // pull character data from the dat file
             SexCG sex = cg.HeritageGroups[(int)character.Heritage].SexList[(int)character.Gender];
-            
+
             character.MotionTableId = sex.MotionTable;
+            character.SetDataIdProperty(PropertyDataId.MotionTable, character.MotionTableId);
+
             character.SoundTableId = sex.SoundTable;
+            character.SetDataIdProperty(PropertyDataId.SoundTable, character.SoundTableId);
+
             character.PhysicsTableId = sex.PhysicsTable;
+            character.SetDataIdProperty(PropertyDataId.PhysicsEffectTable, character.PhysicsTableId);
+
             character.ModelTableId = sex.SetupID;
+            character.SetDataIdProperty(PropertyDataId.Setup, character.ModelTableId);
+
             character.PaletteId = sex.BasePalette;
+            character.SetDataIdProperty(PropertyDataId.PaletteBase, character.PaletteId);
+
             character.CombatTableId = sex.CombatTable;
-            
-            // not sure how to set these.  Optim says they're in the dat
-            // character.SetDataIdProperty(PropertyDataId.EyesTexture, appearance.Nose);
-            // character.SetDataIdProperty(PropertyDataId.NoseTexture, appearance.Nose);
-            // character.SetDataIdProperty(PropertyDataId.MouthTexture, appearance.Mouth);
+            character.SetDataIdProperty(PropertyDataId.CombatTable, character.CombatTableId);
 
-            // character.SetDataIdProperty(PropertyDataId.HairPalette, appearance.HairColor);
-            // character.SetDataIdProperty(PropertyDataId.EyesPalette, appearance.EyeColor);
-            // character.SetDataIdProperty(PropertyDataId.SkinPalette, appearance.EyeColor);
+            // Check the character scale
+            if (sex.Scale != 100u)
+            {
+                character.SetDoubleProperty(PropertyDouble.DefaultScale, (sex.Scale / 100f)); // Scale is stored as a percentage
```

Comment:
I'd REALLY like to see a new "DefaultScale" property on AceCharacter that does this on set.